lib/jekyll/commands/doctor.rb

```diff
@@ -91,6 +94,15 @@ def urls_only_differ_by_case(site)
           urls_only_differ_by_case
         end
 
+        def proper_site_url?(site)
+          url = site.config["url"]
+          [
+            url_exists?(url),
+            url_valid?(url),
```

Comment:
I was thinking of moving `url_exists` out of `proper_site_url?` in a separate test to address Parker's concern.  (We dont wan't to warn users that intentionnaly left site.url blank)